src/webui/api/torrentscontroller.cpp

```diff
@@ -196,6 +196,33 @@ namespace
 
         return QVariantList {dht, pex, lsd};
     }
+
+    BitTorrent::PeerAddress parsePeer(QString peer)
+    {
+        BitTorrent::PeerAddress addr;
+        QStringList ipPort;
+
+        if ((peer[0] == '[') && (peer.indexOf("]:") != -1)) // IPv6
+            ipPort = peer.remove(QChar('[')).split("]:");
```

Comment:
Maybe you can make `peer` const and use something like 
```
socketAddr = peer.mid(1).split("]:"); // socketAddr because you're using the host part of it as well.
```
or inline where you need the host/port part of the socket address but that would be more complicated i think.
```
ipPort = peer.mid(peer.indexOf("]:") + 2);
ipHost = peer.mid(1, (peer.indexOf("]:") - 1));
```
or just rename ipPort to socketAddr or something like that.

Reply:
I've switched to your first example and made `peer` a const ref. Thanks for the suggestion.